02.calendar/cal.js

```diff
@@ -0,0 +1,69 @@
+var lastDayOfMonth = require("date-fns/lastDayOfMonth");
+var getDay = require("date-fns/getDay");
+var argv = require("minimist")(process.argv);
+
+let dateObj = new Date();
+let year = dateObj.getFullYear();
+let month = dateObj.getMonth() + 1;
+let error_msg = "";
+
+// -mオプション
+if ("m" in argv) {
+  if (typeof argv["m"] === "number") {
+    argv["m"] < 1 || argv["m"] > 12
+      ? (error_msg = `cal: ${argv["m"]} is neither a month number (1..12) nor a name`)
+      : (month = argv["m"]);
+  } else {
+    argv["m"] === true
+      ? (error_msg = "cal: option requires an argument -- m")
+      : (error_msg = `cal: ${argv["m"]} is neither a month number (1..12) nor a name`);
+  }
```

Comment:
ちょっとややこしすぎるかなと思うので、
```suggestion
  if (typeof argv["m"] === "number" &&  argv["m"] >= 1 && argv["m"] <= 12)  {
     month = argv["m"];
  } else {
     console.log( `cal: option m need to be a number from 1 to 12`)
     exit(1)
  }
```
とかでどうでしょうか。

Reply:
ifの条件式が複雑になりすぎていたのでご指摘のとおり修正しました。